### PR TITLE
Migrate from Yarn 1 (Classic) to npm

### PR DESCRIPTION
**Pull request by DanielVenable**
Closes [https://github.com/ghostfolio/ghostfolio/issues/3511](https://github.com/ghostfolio/ghostfolio/issues/3511)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -30,26 +30,26 @@ Use `*ngIf="user?.settings?.isExperimentalFeatures"` in HTML template
 
 #### Upgrade
 
-1. Run `yarn nx migrate latest`
-1. Make sure `package.json` changes make sense and then run `yarn install`
-1. Run `yarn nx migrate --run-migrations` (Run `YARN_NODE_LINKER="node-modules" NX_MIGRATE_SKIP_INSTALL=1 yarn nx migrate --run-migrations` due to https://github.com/nrwl/nx/issues/16338)
+1. Run `npx nx migrate latest`
+1. Make sure `package.json` changes make sense and then run `npm install`
+1. Run `npx nx migrate --run-migrations` (Run `YARN_NODE_LINKER="node-modules" NX_MIGRATE_SKIP_INSTALL=1 npx nx migrate --run-migrations` due to https://github.com/nrwl/nx/issues/16338)
```
Comment (dtslvr) [inline DEVELOPMENT.md:35]: The instruction in brackets is probably no longer necessary

File: package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "database:migrate": "prisma migrate deploy",
     "database:push": "prisma db push",
     "database:seed": "prisma db seed",
-    "database:setup": "yarn database:push && yarn database:seed",
+    "database:setup": "npm run database:push && npm database:seed",
```
Comment (dtslvr) [inline package.json:27]: `npm run database:seed` ❓

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for this contribution, @DanielVenable! 😎
I have added a few comments. Can you adapt `git-hooks/pre-commit` as well?

**Review by dtslvr — CHANGES_REQUESTED** [review]
When i run `npm install` locally, I get the following error:

```
ghostfolio % npm install
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: ghostfolio@2.91.0
npm error Found: @angular-devkit/build-angular@18.0.3
npm error node_modules/@angular-devkit/build-angular
npm error   dev @angular-devkit/build-angular@"18.0.5" from the root project
npm error   peer @angular-devkit/build-angular@">= 16.0.0 < 19.0.0" from @nx/angular@19.2.2
npm error   node_modules/@nx/angular
npm error     dev @nx/angular@"19.2.2" from the root project
npm error     @nx/angular@"19.2.2" from @nrwl/angular@19.2.2
npm error     node_modules/@nrwl/angular
npm error       @nrwl/angular@"19.2.2" from @nx/angular@19.2.2
npm error   2 more (jest-preset-angular, ng-extract-i18n-merge)
npm error
npm error Could not resolve dependency:
npm error dev @angular-devkit/build-angular@"18.0.5" from the root project
npm error
npm error Conflicting peer dependency: @angular/compiler@18.0.4
npm error node_modules/@angular/compiler
npm error   peer @angular/compiler@"18.0.4" from @angular/localize@18.0.4
npm error   node_modules/@angular/localize
npm error     dev @angular/localize@"18.0.4" from the root project
npm error     peerOptional @angular/localize@"^18.0.0" from @angular-devkit/build-angular@18.0.5
npm error     node_modules/@angular-devkit/build-angular
npm error       dev @angular-devkit/build-angular@"18.0.5" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

The same happens in the GitHub action. Do you see what's the problem and how to resolve, @DanielVenable?

**Comment by DanielVenable** [comment]
I had the same problem, even before I made any changes. Using --legacy-peer-deps worked.

File: .github/workflows/build-code.yml [hunk 3]
```diff
@@ -24,16 +24,16 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node_version }}
-          cache: 'yarn'
+          cache: 'npm'
 
       - name: Install dependencies
-        run: yarn install --frozen-lockfile
+        run: npm ci
```
Comment (dtslvr) [inline .github/workflows/build-code.yml:30]: Do we need to use `npm ci --legacy-peer-deps` here that the build passes?

**Review by dtslvr — CHANGES_REQUESTED** [review]

File: .github/workflows/build-code.yml (hunk 3 above)
Comment (DanielVenable) [reply .github/workflows/build-code.yml]: That would work.

**Comment by dtslvr** [comment]
Superseded by https://github.com/ghostfolio/ghostfolio/pull/3601

**Pull request closed without merging**